Compilation Error When Building Project (Guess yaml-cpp conflict)

### Short Problem Description:

I'm encountering a compilation error when trying to build the project using the ROS-free setup guide from this repo. I am successfully executing the cmake configuration and afterwards during compilation (build) I get a compilation error which I think has to do with the yaml-cpp dependency.

### System Information:

    Operating System: Ubuntu 22.04
    C++ Compiler: GNU C++ 11.4.0 (C++ 17 supported)
    CMake Version: 3.22.1
    yaml-cpp Version: 0.7.0 (on my system)

#### Note:

A friend of mine did it with 0.6.2 yaml-cpp version and it worked. Since in the config it uses the master (which is currently 0.8.0) This means that the Fetch is being ignored and the system wide installed package is being used. 

### Steps to Reproduce:

1. Clone the project and navigate to the project directory.
2. mkdir -p build/Debug
3. **Run** cd build/Debug && cmake -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTS=ON ../..*
4. **Run** cmake --build . --config Debug --target all -j && cd ..

### Observed Behavior during CMake Configuration

This is the output after executing the cmake configuration command (**Step 3**):

```
-- The CXX compiler identification is GNU 11.4.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.74.0/BoostConfig.cmake (found version "1.74.0")  
-- Found OpenCV: /usr (found suitable version "4.5.4", minimum required is "4") 
-- The C compiler identification is GNU 11.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Performing Test EIGEN_COMPILER_SUPPORT_CPP11
-- Performing Test EIGEN_COMPILER_SUPPORT_CPP11 - Success
-- Performing Test standard_math_library_linked_to_automatically
-- Performing Test standard_math_library_linked_to_automatically - Success
-- Standard libraries to link to explicitly: none
-- Performing Test COMPILER_SUPPORT_WERROR
-- Performing Test COMPILER_SUPPORT_WERROR - Success
-- Performing Test COMPILER_SUPPORT_pedantic
-- Performing Test COMPILER_SUPPORT_pedantic - Success
-- Performing Test COMPILER_SUPPORT_Wall
-- Performing Test COMPILER_SUPPORT_Wall - Success
-- Performing Test COMPILER_SUPPORT_Wextra
-- Performing Test COMPILER_SUPPORT_Wextra - Success
-- Performing Test COMPILER_SUPPORT_Wundef
-- Performing Test COMPILER_SUPPORT_Wundef - Success
-- Performing Test COMPILER_SUPPORT_Wcastalign
-- Performing Test COMPILER_SUPPORT_Wcastalign - Success
-- Performing Test COMPILER_SUPPORT_Wcharsubscripts
-- Performing Test COMPILER_SUPPORT_Wcharsubscripts - Success
-- Performing Test COMPILER_SUPPORT_Wnonvirtualdtor
-- Performing Test COMPILER_SUPPORT_Wnonvirtualdtor - Success
-- Performing Test COMPILER_SUPPORT_Wunusedlocaltypedefs
-- Performing Test COMPILER_SUPPORT_Wunusedlocaltypedefs - Success
-- Performing Test COMPILER_SUPPORT_Wpointerarith
-- Performing Test COMPILER_SUPPORT_Wpointerarith - Success
-- Performing Test COMPILER_SUPPORT_Wwritestrings
-- Performing Test COMPILER_SUPPORT_Wwritestrings - Success
-- Performing Test COMPILER_SUPPORT_Wformatsecurity
-- Performing Test COMPILER_SUPPORT_Wformatsecurity - Success
-- Performing Test COMPILER_SUPPORT_Wshorten64to32
-- Performing Test COMPILER_SUPPORT_Wshorten64to32 - Failed
-- Performing Test COMPILER_SUPPORT_Wlogicalop
-- Performing Test COMPILER_SUPPORT_Wlogicalop - Success
-- Performing Test COMPILER_SUPPORT_Wenumconversion
-- Performing Test COMPILER_SUPPORT_Wenumconversion - Success
-- Performing Test COMPILER_SUPPORT_Wcpp11extensions
-- Performing Test COMPILER_SUPPORT_Wcpp11extensions - Failed
-- Performing Test COMPILER_SUPPORT_Wdoublepromotion
-- Performing Test COMPILER_SUPPORT_Wdoublepromotion - Success
-- Performing Test COMPILER_SUPPORT_Wshadow
-- Performing Test COMPILER_SUPPORT_Wshadow - Success
-- Performing Test COMPILER_SUPPORT_Wnopsabi
-- Performing Test COMPILER_SUPPORT_Wnopsabi - Success
-- Performing Test COMPILER_SUPPORT_Wnovariadicmacros
-- Performing Test COMPILER_SUPPORT_Wnovariadicmacros - Success
-- Performing Test COMPILER_SUPPORT_Wnolonglong
-- Performing Test COMPILER_SUPPORT_Wnolonglong - Success
-- Performing Test COMPILER_SUPPORT_fnochecknew
-- Performing Test COMPILER_SUPPORT_fnochecknew - Success
-- Performing Test COMPILER_SUPPORT_fnocommon
-- Performing Test COMPILER_SUPPORT_fnocommon - Success
-- Performing Test COMPILER_SUPPORT_fstrictaliasing
-- Performing Test COMPILER_SUPPORT_fstrictaliasing - Success
-- Performing Test COMPILER_SUPPORT_wd981
-- Performing Test COMPILER_SUPPORT_wd981 - Failed
-- Performing Test COMPILER_SUPPORT_wd2304
-- Performing Test COMPILER_SUPPORT_wd2304 - Failed
-- Performing Test COMPILER_SUPPORT_STRICTANSI
-- Performing Test COMPILER_SUPPORT_STRICTANSI - Failed
-- Performing Test COMPILER_SUPPORT_Qunusedarguments
-- Performing Test COMPILER_SUPPORT_Qunusedarguments - Failed
-- Performing Test COMPILER_SUPPORT_ansi
-- Performing Test COMPILER_SUPPORT_ansi - Success
-- Performing Test COMPILER_SUPPORT_OPENMP
-- Performing Test COMPILER_SUPPORT_OPENMP - Success
-- Looking for a Fortran compiler
-- Looking for a Fortran compiler - /usr/bin/f95
-- The Fortran compiler identification is GNU 11.4.0
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Check for working Fortran compiler: /usr/bin/f95 - skipped
-- Found CHOLMOD: /usr/include/suitesparse  
-- Found UMFPACK: /usr/include/suitesparse  
-- Found KLU: /usr/include/suitesparse  
-- Performing Test SUPERLU_HAS_GLOBAL_MEM_USAGE_T
-- Performing Test SUPERLU_HAS_GLOBAL_MEM_USAGE_T - Success
-- Performing Test SUPERLU_HAS_CLEAN_ENUMS
-- Performing Test SUPERLU_HAS_CLEAN_ENUMS - Success
-- Performing Test SUPERLU_HAS_GLOBALLU_T
-- Performing Test SUPERLU_HAS_GLOBALLU_T - Success
-- Found SuperLU: /usr/include/superlu (found suitable version "5.0", minimum required is "4.0") 
-- Checking for one of the modules 'hwloc'
-- Performing Test HAVE_HWLOC_PARENT_MEMBER
-- Performing Test HAVE_HWLOC_PARENT_MEMBER - Success
-- Performing Test HAVE_HWLOC_CACHE_ATTR
-- Performing Test HAVE_HWLOC_CACHE_ATTR - Success
-- Performing Test HAVE_HWLOC_OBJ_PU
-- Performing Test HAVE_HWLOC_OBJ_PU - Success
-- Looking for hwloc_bitmap_free in hwloc
-- Looking for hwloc_bitmap_free in hwloc - found
-- Looking for Fortran sgemm
-- Looking for Fortran sgemm - found
-- Performing Test SCOTCH_Num_4
-- Performing Test SCOTCH_Num_4 - Failed
-- Performing Test SCOTCH_Num_8
-- Performing Test SCOTCH_Num_8 - Failed
-- A version of Pastix has been found but pastix_nompi.h does not exist in the include directory. Because Eigen tests require a version without MPI, we disable the Pastix backend.
-- Found SPQR: /usr/include/suitesparse  
-- Found unsuitable Qt version "5.15.3" from /usr/bin/qmake
-- Performing Test COMPILER_SUPPORT_FASTMATH
-- Performing Test COMPILER_SUPPORT_FASTMATH - Success
-- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.74.0/BoostConfig.cmake (found suitable version "1.74.0", minimum required is "1.53.0")  
-- Found Threads: TRUE  
-- Could NOT find GoogleHash (missing: GOOGLEHASH_COMPILE) 
-- Found Adolc: /usr/include  
-- Found MPFR: /usr/include (Required is at least version "1.0.0") 
-- Found GMP: /usr/include/x86_64-linux-gnu  
-- Could NOT find MPREAL (missing: MPREAL_INCLUDES MPREAL_VERSION_OK MPREAL_WORKS) (Required is at least version "1.0.0")
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2") 
-- Found FFTW: /usr/include  
-- Found CUDA: /usr (found suitable version "11.5", minimum required is "7.0") 
-- Found unsuitable Qt version "5.15.3" from /usr/bin/qmake
-- Qt4 not found, so disabling the mandelbrot and opengl demos
-- A version of Pastix has been found but pastix_nompi.h does not exist in the include directory. Because Eigen tests require a version without MPI, we disable the Pastix backend.
-- ************************************************************
-- ***    Eigen's unit tests configuration summary          ***
-- ************************************************************
-- 
-- Build type:        Debug
-- Build site:        user
-- Build string:      linux-6.8.0-48-generic-_-11.4.0-sse2-64bit-cxx11
-- Enabled backends:  CHOLMOD,  UMFPACK,  KLU,  SuperLU,  SPQR,  Boost.Multiprecision,  Adolc,  fftw, 
-- Disabled backends: PaStiX,  METIS,  Qt4 support,  GoogleHash,  MPFR C++,  OpenGL, 
-- Default order:     Column-major
-- Maximal matrix/vector size: 320
-- SSE2:              Using architecture defaults
-- SSE3:              Using architecture defaults
-- SSSE3:             Using architecture defaults
-- SSE4.1:            Using architecture defaults
-- SSE4.2:            Using architecture defaults
-- AVX:               Using architecture defaults
-- AVX2:              Using architecture defaults
-- FMA:               Using architecture defaults
-- AVX512:            Using architecture defaults
-- AVX512DQ:          Using architecture defaults
-- Altivec:           Using architecture defaults
-- VSX:               Using architecture defaults
-- MIPS MSA:          Using architecture defaults
-- ARM NEON:          Using architecture defaults
-- ARMv8 NEON:        Using architecture defaults
-- S390X ZVECTOR:     Using architecture defaults
-- C++11:             ON
-- SYCL:              OFF
-- CUDA:              OFF
-- HIP:               OFF
-- 
CXX:               /usr/bin/c++
 CXX_VERSION:       c++ (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
 CXX_FLAGS:          -std=c++11 -pedantic -Wall -Wextra -Wundef -Wcast-align -Wchar-subscripts -Wnon-virtual-dtor -Wunused-local-typedefs -Wpointer-arith -Wwrite-strings -Wformat-security -Wlogical-op -Wenum-conversion -Wdouble-promotion -Wshadow -Wno-psabi -Wno-variadic-macros -Wno-long-long -fno-check-new -fno-common -fstrict-aliasing -ansi
 Sparse lib flags:   ;/usr/lib/x86_64-linux-gnu/libcholmod.so;/usr/lib/x86_64-linux-gnu/libamd.so;/usr/lib/x86_64-linux-gnu/libcolamd.so;/usr/lib/x86_64-linux-gnu/libcamd.so;/usr/lib/x86_64-linux-gnu/libccolamd.so;eigen_blas;eigen_lapack;/usr/lib/x86_64-linux-gnu/libumfpack.so;/usr/lib/x86_64-linux-gnu/libcolamd.so;/usr/lib/x86_64-linux-gnu/libamd.so;/usr/lib/x86_64-linux-gnu/libcholmod.so;eigen_blas;/usr/lib/x86_64-linux-gnu/libklu.so;/usr/lib/x86_64-linux-gnu/libcolamd.so;/usr/lib/x86_64-linux-gnu/libamd.so;/usr/lib/x86_64-linux-gnu/libbtf.so;eigen_blas;/usr/lib/x86_64-linux-gnu/libsuperlu.so;eigen_blas;/usr/lib/x86_64-linux-gnu/libspqr.so;/usr/lib/x86_64-linux-gnu/libcholmod.so;/usr/lib/x86_64-linux-gnu/libcholmod.so;/usr/lib/x86_64-linux-gnu/libamd.so;/usr/lib/x86_64-linux-gnu/libcolamd.so;/usr/lib/x86_64-linux-gnu/libcamd.so;/usr/lib/x86_64-linux-gnu/libccolamd.so;eigen_lapack;eigen_blas

-- ************************************************************
-- 
-- Configured Eigen 3.4.0
-- 
-- Available targets (use: make TARGET):
-- ---------+--------------------------------------------------------------
-- Target   |   Description
-- ---------+--------------------------------------------------------------
-- install  | Install Eigen. Headers will be installed to:
--          |     <CMAKE_INSTALL_PREFIX>/<INCLUDE_INSTALL_DIR>
--          |   Using the following values:
--          |     CMAKE_INSTALL_PREFIX: /usr/local
--          |     INCLUDE_INSTALL_DIR:  include/eigen3
--          |   Change the install location of Eigen headers using:
--          |     cmake . -DCMAKE_INSTALL_PREFIX=yourprefix
--          |   Or:
--          |     cmake . -DINCLUDE_INSTALL_DIR=yourdir
-- doc      | Generate the API documentation, requires Doxygen & LaTeX
-- check    | Build and run the unit-tests. Read this page:
--          |   http://eigen.tuxfamily.org/index.php?title=Tests
-- blas     | Build BLAS library (not the same thing as Eigen)
-- uninstall| Remove files installed by the install target
-- ---------+--------------------------------------------------------------
-- 
-- Performing Test COMPILER_SUPPORTS_CXX17
-- Performing Test COMPILER_SUPPORTS_CXX17 - Success
-- Performing Test COMPILER_SUPPORTS_O0
-- Performing Test COMPILER_SUPPORTS_O0 - Success
-- Performing Test COMPILER_SUPPORTS_G3
-- Performing Test COMPILER_SUPPORTS_G3 - Success
-- Performing Test COMPILER_SUPPORTS_WALL
-- Performing Test COMPILER_SUPPORTS_WALL - Success
-- Performing Test COMPILER_SUPPORTS_WEXTRA
-- Performing Test COMPILER_SUPPORTS_WEXTRA - Success
-- Performing Test COMPILER_SUPPORTS_WERROR
-- Performing Test COMPILER_SUPPORTS_WERROR - Success
-- Performing Test COMPILER_SUPPORTS_WUNINITIALIZED
-- Performing Test COMPILER_SUPPORTS_WUNINITIALIZED - Success
-- Performing Test COMPILER_SUPPORTS_WMAYBEUNINITIALIZED
-- Performing Test COMPILER_SUPPORTS_WMAYBEUNINITIALIZED - Success
-- Performing Test COMPILER_SUPPORTS_PEDANTIC
-- Performing Test COMPILER_SUPPORTS_PEDANTIC - Success
-- Performing Test COMPILER_SUPPORTS_FNOOMITFRAMEPOINTER
-- Performing Test COMPILER_SUPPORTS_FNOOMITFRAMEPOINTER - Success
-- COMPILER: /usr/bin/c++
-- CMAKE_BUILD_TYPE: Debug
-- Eigen VER: 
-- Lie++ VER: 
-- Yaml-cpp VER: 
-- Boost VER: 1.74.0
-- OpenCV VER: 4.5.4
-- Building native C++
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTS


-- Build files have been written to: /home/user/git/MSCEqF/build/Debug
`


 that includes yaml-cpp as a dependency. The issue appears to occur during the cmake configuration and build steps.
```
 ### Observed Behavior: During Compilation
 
 And the error occurs after executing the following line: cmake --build . --config Debug --target all -j && cd ../..
 
```
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/binary.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/depthguard.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/directives.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/contrib/graphbuilder.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/emitfromevents.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/contrib/graphbuilderadapter.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/emitterstate.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/convert.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/exp.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/exceptions.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/emit.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/fptostring.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/emitter.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/emitterutils.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/node_data.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/nodeevents.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/node.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/nodebuilder.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/memory.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/null.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/parse.cpp.o
[  0%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/ostream_wrapper.cpp.o
[ 50%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/regex_yaml.cpp.o
[ 50%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/parser.cpp.o
[ 50%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/scanscalar.cpp.o
[ 50%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/scanner.cpp.o
[ 50%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/scantag.cpp.o
[ 50%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/scantoken.cpp.o
[ 50%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/simplekey.cpp.o
[ 50%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/singledocparser.cpp.o
[ 50%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/tag.cpp.o
[ 50%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/stream.cpp.o
[ 50%] Linking CXX static library libyaml-cppd.a
[ 50%] Built target yaml-cpp
[ 50%] Building CXX object _deps/yaml-cpp-build/util/CMakeFiles/yaml-cpp-parse.dir/parse.cpp.o
[ 50%] Building CXX object CMakeFiles/msceqf_lib.dir/source/msceqf/msceqf.cpp.o
[ 50%] Building CXX object CMakeFiles/msceqf_lib.dir/source/msceqf/filter/updater/zero_velocity_updater.cpp.o
[ 50%] Building CXX object CMakeFiles/msceqf_lib.dir/source/msceqf/filter/updater/updater.cpp.o
[ 50%] Building CXX object CMakeFiles/msceqf_lib.dir/source/msceqf/system/system.cpp.o
[ 50%] Building CXX object CMakeFiles/msceqf_lib.dir/source/msceqf/options/msceqf_option_parser.cpp.o
[ 50%] Building CXX object _deps/yaml-cpp-build/util/CMakeFiles/yaml-cpp-sandbox.dir/sandbox.cpp.o
[100%] Building CXX object CMakeFiles/msceqf_lib.dir/source/msceqf/state/state.cpp.o
[100%] Building CXX object CMakeFiles/msceqf_lib.dir/source/msceqf/filter/propagator/propagator.cpp.o
[100%] Building CXX object CMakeFiles/msceqf_lib.dir/source/msceqf/symmetry/symmetry.cpp.o
[100%] Building CXX object _deps/yaml-cpp-build/util/CMakeFiles/yaml-cpp-read.dir/read.cpp.o
[100%] Building CXX object CMakeFiles/msceqf_lib.dir/source/msceqf/filter/checker/checker.cpp.o
[100%] Building CXX object CMakeFiles/msceqf_lib.dir/source/msceqf/filter/initializer/static_initializer.cpp.o
[100%] Building CXX object CMakeFiles/msceqf_lib.dir/source/msceqf/filter/updater/updater_helper.cpp.o
[100%] Building CXX object CMakeFiles/msceqf_lib.dir/source/vision/tracker.cpp.o
[100%] Building CXX object CMakeFiles/msceqf_lib.dir/source/vision/track_manager.cpp.o
[100%] Building CXX object CMakeFiles/msceqf_lib.dir/source/vision/camera.cpp.o
[100%] Linking CXX executable parse
[100%] Built target yaml-cpp-parse
[100%] Linking CXX executable sandbox
[100%] Built target yaml-cpp-sandbox
[100%] Linking CXX executable read
[100%] Built target yaml-cpp-read
In file included from /home/usr/git/state_estimation/MSCEqF/source/msceqf/options/msceqf_option_parser.cpp:12:
/home/usr/git/state_estimation/MSCEqF/include/msceqf/options/msceqf_option_parser.hpp: In member function ‘bool msceqf::OptionParser::read(msceqf::Quaternion&, const string&)’:
/home/usr/git/state_estimation/MSCEqF/include/msceqf/options/msceqf_option_parser.hpp:134:27: error: invalid ‘static_cast’ from type ‘std::__cxx11::basic_ostringstream<char>’ to type ‘std::ostringstream&’ {aka ‘std::__cxx11::basic_ostringstream<char>&’}
  134 |                           static_cast<std::ostringstream&>(std::ostringstream() << q).str());
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/usr/git/state_estimation/MSCEqF/include/msceqf/options/msceqf_option_parser.hpp: In instantiation of ‘void msceqf::OptionParser::readDefault(T&, const Convertible&, const string&) [with T = int; Convertible = int; std::string = std::__cxx11::basic_string<char>]’:
/home/usr/git/state_estimation/MSCEqF/source/msceqf/options/msceqf_option_parser.cpp:27:14:   required from here
/home/usr/git/state_estimation/MSCEqF/include/msceqf/options/msceqf_option_parser.hpp:171:27: error: invalid ‘static_cast’ from type ‘std::__cxx11::basic_ostringstream<char>’ to type ‘std::ostringstream&’ {aka ‘std::__cxx11::basic_ostringstream<char>&’}
  171 |                           static_cast<std::ostringstream&>(std::ostringstream() << p).str());
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/usr/git/state_estimation/MSCEqF/include/msceqf/options/msceqf_option_parser.hpp: In instantiation of ‘void msceqf::OptionParser::readDefault(T&, const Convertible&, const string&) [with T = bool; Convertible = bool; std::string = std::__cxx11::basic_string<char>]’:
/home/usr/git/state_estimation/MSCEqF/source/msceqf/options/msceqf_option_parser.cpp:35:14:   required from here
/home/usr/git/state_estimation/MSCEqF/include/msceqf/options/msceqf_option_parser.hpp:171:27: error: invalid ‘static_cast’ from type ‘std::__cxx11::basic_ostringstream<char>’ to type ‘std::ostringstream&’ {aka ‘std::__cxx11::basic_ostringstream<char>&’}
/home/usr/git/state_estimation/MSCEqF/include/msceqf/options/msceqf_option_parser.hpp: In instantiation of ‘void msceqf::OptionParser::readDefault(T&, const Convertible&, const string&) [with T = double; Convertible = double; std::string = std::__cxx11::basic_string<char>]’:
/home/usr/git/state_estimation/MSCEqF/source/msceqf/options/msceqf_option_parser.cpp:42:14:   required from here
/home/usr/git/state_estimation/MSCEqF/include/msceqf/options/msceqf_option_parser.hpp:171:27: error: invalid ‘static_cast’ from type ‘std::__cxx11::basic_ostringstream<char>’ to type ‘std::ostringstream&’ {aka ‘std::__cxx11::basic_ostringstream<char>&’}
/home/usr/git/state_estimation/MSCEqF/include/msceqf/options/msceqf_option_parser.hpp: In instantiation of ‘void msceqf::OptionParser::readDefault(T&, const Convertible&, const string&) [with T = unsigned int; Convertible = int; std::string = std::__cxx11::basic_string<char>]’:
/home/usr/git/state_estimation/MSCEqF/source/msceqf/options/msceqf_option_parser.cpp:43:14:   required from here
/home/usr/git/state_estimation/MSCEqF/include/msceqf/options/msceqf_option_parser.hpp:171:27: error: invalid ‘static_cast’ from type ‘std::__cxx11::basic_ostringstream<char>’ to type ‘std::ostringstream&’ {aka ‘std::__cxx11::basic_ostringstream<char>&’}
/home/usr/git/state_estimation/MSCEqF/include/msceqf/options/msceqf_option_parser.hpp: In instantiation of ‘void msceqf::OptionParser::readDefault(T&, const Convertible&, const string&) [with T = long unsigned int; Convertible = int; std::string = std::__cxx11::basic_string<char>]’:
/home/usr/git/state_estimation/MSCEqF/source/msceqf/options/msceqf_option_parser.cpp:99:14:   required from here
/home/usr/git/state_estimation/MSCEqF/include/msceqf/options/msceqf_option_parser.hpp:171:27: error: invalid ‘static_cast’ from type ‘std::__cxx11::basic_ostringstream<char>’ to type ‘std::ostringstream&’ {aka ‘std::__cxx11::basic_ostringstream<char>&’}
/home/usr/git/state_estimation/MSCEqF/include/msceqf/options/msceqf_option_parser.hpp: In instantiation of ‘void msceqf::OptionParser::readDefault(T&, const Convertible&, const string&) [with T = double; Convertible = int; std::string = std::__cxx11::basic_string<char>]’:
/home/usr/git/state_estimation/MSCEqF/source/msceqf/options/msceqf_option_parser.cpp:142:14:   required from here
/home/usr/git/state_estimation/MSCEqF/include/msceqf/options/msceqf_option_parser.hpp:171:27: error: invalid ‘static_cast’ from type ‘std::__cxx11::basic_ostringstream<char>’ to type ‘std::ostringstream&’ {aka ‘std::__cxx11::basic_ostringstream<char>&’}
/home/usr/git/state_estimation/MSCEqF/include/msceqf/options/msceqf_option_parser.hpp: In instantiation of ‘bool msceqf::OptionParser::read(msceqf::Matrix<Rows, Cols>&, const string&) [with int Rows = 4; int Cols = 4; msceqf::Matrix<Rows, Cols> = Eigen::Matrix<double, 4, 4>; std::string = std::__cxx11::basic_string<char>]’:

```

I changed the yaml-cpp fetch as following:

```
FetchContent_Declare(
    yaml-cpp
    GIT_REPOSITORY  https://github.com/jbeder/yaml-cpp.git
    GIT_TAG         yaml-cpp-0.6.2
    GIT_SHALLOW     TRUE
    GIT_PROGRESS    TRUE
)
```

This leads during the configuration to the following error:

```
CMake Deprecation Warning at build/Release/_deps/yaml-cpp-src/test/gtest-1.8.0/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Warning (dev) at build/Release/_deps/yaml-cpp-src/test/gtest-1.8.0/CMakeLists.txt:3 (project):
  Policy CMP0048 is not set: project() command manages VERSION variables.
  Run "cmake --help-policy CMP0048" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The following variable(s) would be set to empty:

    PROJECT_VERSION
    PROJECT_VERSION_MAJOR
    PROJECT_VERSION_MINOR
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at build/Release/_deps/yaml-cpp-src/test/gtest-1.8.0/googlemock/CMakeLists.txt:40 (project):
  Policy CMP0048 is not set: project() command manages VERSION variables.
  Run "cmake --help-policy CMP0048" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The following variable(s) would be set to empty:

    PROJECT_VERSION
    PROJECT_VERSION_MAJOR
    PROJECT_VERSION_MINOR
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Deprecation Warning at build/Release/_deps/yaml-cpp-src/test/gtest-1.8.0/googlemock/CMakeLists.txt:41 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Warning (dev) at build/Release/_deps/yaml-cpp-src/test/gtest-1.8.0/googletest/CMakeLists.txt:47 (project):
  Policy CMP0048 is not set: project() command manages VERSION variables.
  Run "cmake --help-policy CMP0048" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The following variable(s) would be set to empty:

    PROJECT_VERSION
    PROJECT_VERSION_MAJOR
    PROJECT_VERSION_MINOR
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Deprecation Warning at build/Release/_deps/yaml-cpp-src/test/gtest-1.8.0/googletest/CMakeLists.txt:48 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

```


-------------------------------------------------------------------------------------------------------------------------------------

and as you said 

```
FetchContent_Declare(
    yaml-cpp
    GIT_REPOSITORY  https://github.com/jbeder/yaml-cpp.git
    GIT_TAG         0.8.0
    GIT_SHALLOW     TRUE
    GIT_PROGRESS    TRUE
)
```

I get the following output ...
```

-- ************************************************************
-- 
-- Configured Eigen 3.4.0
-- 
-- Available targets (use: make TARGET):
-- ---------+--------------------------------------------------------------
-- Target   |   Description
-- ---------+--------------------------------------------------------------
-- install  | Install Eigen. Headers will be installed to:
--          |     <CMAKE_INSTALL_PREFIX>/<INCLUDE_INSTALL_DIR>
--          |   Using the following values:
--          |     CMAKE_INSTALL_PREFIX: /usr/local
--          |     INCLUDE_INSTALL_DIR:  include/eigen3
--          |   Change the install location of Eigen headers using:
--          |     cmake . -DCMAKE_INSTALL_PREFIX=yourprefix
--          |   Or:
--          |     cmake . -DINCLUDE_INSTALL_DIR=yourdir
-- doc      | Generate the API documentation, requires Doxygen & LaTeX
-- check    | Build and run the unit-tests. Read this page:
--          |   http://eigen.tuxfamily.org/index.php?title=Tests
-- blas     | Build BLAS library (not the same thing as Eigen)
-- uninstall| Remove files installed by the install target
-- ---------+--------------------------------------------------------------
-- 
CMake Error at build/Release/_deps/eigen-src/CMakeLists.txt:648 (add_custom_target):
  add_custom_target cannot create target "uninstall" because another target
  with the same name already exists.  The existing target is a custom target
  created in source directory
  "/home/pierno/git/state_estimation/MSCEqF/build/Release/_deps/yaml-cpp-src".
  See documentation for policy CMP0002 for more details.


-- Performing Test COMPILER_SUPPORTS_MARCH=NATIVE
-- Performing Test COMPILER_SUPPORTS_MARCH=NATIVE - Success
-- COMPILER: /usr/bin/c++
-- CMAKE_BUILD_TYPE: Release
-- Eigen VER: 
-- Lie++ VER: 
-- Yaml-cpp VER: 
-- Boost VER: 1.74.0
-- OpenCV VER: 4.5.4
-- Building native C++
-- Configuring incomplete, errors occurred!
See also "/home/pierno/git/state_estimation/MSCEqF/build/Release/CMakeFiles/CMakeOutput.log".
See also "/home/pierno/git/state_estimation/MSCEqF/build/Release/CMakeFiles/CMakeError.log".

```

Here is the CMakeOutput.log and CMakeError.log


[CMakeOutput.log](https://github.com/user-attachments/files/17750034/CMakeOutput.log)
[CMakeError.log](https://github.com/user-attachments/files/17750036/CMakeError.log)

--- a/docker/Dockerfile_ros2
+++ b/docker/Dockerfile_ros2
@@ -1,4 +1,5 @@
-FROM osrf/ros:galactic-desktop
+#FROM osrf/ros:galactic-desktop
+FROM osrf/ros:iron-desktop
 
 RUN apt-get update && apt-get install -y nano git
 

--- a/include/msceqf/options/msceqf_option_parser.hpp
+++ b/include/msceqf/options/msceqf_option_parser.hpp
@@ -25,25 +25,25 @@
 
 namespace msceqf
 {
-class OptionParser
-{
- public:
-  /**
+  class OptionParser
+  {
+  public:
+    /**
    * @brief Option parser constructor
    *
    * @param filepath Parameter file
    */
-  OptionParser(const std::string& filepath);
+    OptionParser(const std::string &filepath);
 
-  /**
+    /**
    * @brief Parse oprion and create MSCEqFOptions struct
    *
    * @return MSCEqFOptions
    */
-  MSCEqFOptions parseOptions();
+    MSCEqFOptions parseOptions();
 
- private:
-  /**
+  private:
+    /**
    * @brief Read a parameter from the YAML file and store it in a matrix or vector.
    * A vector needs to be specified as [a, b, c, ..., z] in the YAMl file.
    * A matrix needs to be specified as a a list of vectors, each representing a row of the matrix.
@@ -55,64 +55,64 @@ class OptionParser
    * @param param Parameter to be read
    * @return true if parameter correctly parsed, false otherwise
    */
-  template <int Rows, int Cols>
-  [[nodiscard]] bool read(Matrix<Rows, Cols>& x, const std::string& param)
-  {
-    if (node_[param])
+    template <int Rows, int Cols>
+    [[nodiscard]] bool read(Matrix<Rows, Cols> &x, const std::string &param)
     {
-      using vector = std::vector<fp>;
-      using vectorvector = std::vector<std::vector<fp>>;
+      if (node_[param])
+      {
+        using vector = std::vector<fp>;
+        using vectorvector = std::vector<std::vector<fp>>;
 
-      int rows = Rows;
-      int cols = Cols;
+        int rows = Rows;
+        int cols = Cols;
 
-      if constexpr (Rows == 1 || Cols == 1)
-      {
-        vector vec = node_[param].as<vector>();
-        if constexpr (Rows == -1)
+        if constexpr (Rows == 1 || Cols == 1)
         {
-          rows = vec.size();
+          vector vec = node_[param].as<vector>();
+          if constexpr (Rows == -1)
+          {
+            rows = vec.size();
+          }
+          else if (Cols == -1)
+          {
+            cols = vec.size();
+          }
+          x = Map<Matrix<Rows, Cols>>(vec.data(), rows, cols);
         }
-        else if (Cols == -1)
+        else
         {
-          cols = vec.size();
+          vectorvector mat = node_[param].as<vectorvector>();
+          vector vec = utils::flatten(mat);
+          if constexpr (Rows == 1 && Cols == 1)
+          {
+            rows = mat.size();
+            cols = vec.size() / mat.size();
+          }
+          x = Map<Matrix<Rows, Cols, Eigen::RowMajor>>(vec.data(), rows, cols);
         }
-        x = Map<Matrix<Rows, Cols>>(vec.data(), rows, cols);
-      }
-      else
-      {
-        vectorvector mat = node_[param].as<vectorvector>();
-        vector vec = utils::flatten(mat);
-        if constexpr (Rows == 1 && Cols == 1)
+        if constexpr (Rows == 1)
         {
-          rows = mat.size();
-          cols = vec.size() / mat.size();
+          utils::Logger::info("Parameter: [" + param + "] found. Option set to:" +
+                              (std::ostringstream{} << x).str());
         }
-        x = Map<Matrix<Rows, Cols, Eigen::RowMajor>>(vec.data(), rows, cols);
-      }
-      if constexpr (Rows == 1)
-      {
-        utils::Logger::info("Parameter: [" + param + "] found. Option set to:" +
-                            static_cast<std::ostringstream&>(std::ostringstream() << x).str());
-      }
-      else if (Cols == 1)
-      {
-        utils::Logger::info("Parameter: [" + param + "] found. Option set to: " +
-                            static_cast<std::ostringstream&>(std::ostringstream() << x.transpose()).str());
-      }
-      else
-      {
-        utils::Logger::info("Parameter: [" + param + "] found. Option set to: \n" +
-                            static_cast<std::ostringstream&>(std::ostringstream() << x).str());
-      }
+        else if (Cols == 1)
+        {
+          utils::Logger::info("Parameter: [" + param + "] found. Option set to: " +
+                              (std::ostringstream{} << x.transpose()).str());
+        }
+        else
+        {
+          utils::Logger::info("Parameter: [" + param + "] found. Option set to: \n" +
+                              (std::ostringstream{} << x).str());
+        }
 
-      return true;
+        return true;
+      }
+      utils::Logger::warn("Parameter: [" + param + "] not found");
+      return false;
     }
-    utils::Logger::warn("Parameter: [" + param + "] not found");
-    return false;
-  }
 
-  /**
+    /**
    * @brief Read a parameter from the YAML file and store it in a quaternion.
    * The quaternion needs to be specified as [x, y, z, w] in the YAML file.
    * the function perform quaternion normalization.
@@ -122,23 +122,23 @@ class OptionParser
    * @param parameter Parameter to be read
    * @return true if parameter correctly parsed, false otherwise
    */
-  template <typename Scalar>
-  [[nodiscard]] bool read(Quaternion& q, const std::string& param)
-  {
-    if (node_[param])
+    template <typename Scalar>
+    [[nodiscard]] bool read(Quaternion &q, const std::string &param)
     {
-      using vector = std::vector<fp>;
-      vector vec = node_[param].as<vector>();
-      q = Quaternion(vec).normalize();
-      utils::Logger::info("Parameter: [" + param + "] found. Option set to: \n" +
-                          static_cast<std::ostringstream&>(std::ostringstream() << q).str());
-      return true;
+      if (node_[param])
+      {
+        using vector = std::vector<fp>;
+        vector vec = node_[param].as<vector>();
+        q = Quaternion(vec).normalize();
+        utils::Logger::info("Parameter: [" + param + "] found. Option set to: \n" +
+                            (std::ostringstream{} << q).str());
+        return true;
+      }
+      utils::Logger::warn("Parameter: [" + param + "] not found");
+      return false;
     }
-    utils::Logger::warn("Parameter: [" + param + "] not found");
-    return false;
-  }
 
-  /**
+    /**
    * @brief Read a parameter from the YAML file and store it in a variable of type T.
    *
    * @tparam T type of the variable where the parameter has to be stored in
@@ -146,33 +146,33 @@ class OptionParser
    * @param param
    * @return true if parameter correctly parsed, false otherwise
    */
-  template <typename T>
-  [[nodiscard]] bool read(T& p, const std::string& param)
-  {
-    if (node_[param])
+    template <typename T>
+    [[nodiscard]] bool read(T &p, const std::string &param)
     {
-      p = node_[param].as<T>();
-      utils::Logger::info("Parameter: [" + param + "] found. Option set to: " +
-                          static_cast<std::ostringstream&>(std::ostringstream() << p).str());
-      return true;
+      if (node_[param])
+      {
+        p = node_[param].as<T>();
+        utils::Logger::info("Parameter: [" + param + "] found. Option set to: " +
+                            (std::ostringstream{} << p).str());
+        return true;
+      }
+      utils::Logger::warn("Parameter: [" + param + "] not found");
+      return false;
     }
-    utils::Logger::warn("Parameter: [" + param + "] not found");
-    return false;
-  }
 
-  template <typename T, typename Convertible>
-  void readDefault(T& p, const Convertible& def, const std::string& param)
-  {
-    static_assert(std::is_convertible_v<Convertible, T>);
-    if (!read(p, param))
+    template <typename T, typename Convertible>
+    void readDefault(T &p, const Convertible &def, const std::string &param)
     {
-      p = def;
-      utils::Logger::warn("Parameter: [" + param + "] set to default value: " +
-                          static_cast<std::ostringstream&>(std::ostringstream() << p).str());
+      static_assert(std::is_convertible_v<Convertible, T>);
+      if (!read(p, param))
+      {
+        p = def;
+        utils::Logger::warn("Parameter: [" + param + "] set to default value: " +
+                            (std::ostringstream{} << p).str());
+      }
     }
-  }
 
-  /**
+    /**
    * @brief Parse the camera parmaeters, including extrinsics and intrinsics from the YAML file.
    * The camera parameters has to be defined according to Kalibr convention
    *
@@ -185,53 +185,53 @@ class OptionParser
    * @param mask
    * @param mask_type
    */
-  void parseCameraParameters(SE3& extrinsics,
-                             In& intrinsics,
-                             DistortionModel& distortion_model,
-                             VectorX& distortion_coefficients,
-                             Vector2& resolution,
-                             fp& timeshift_cam_imu,
-                             cv::Mat& mask,
-                             MaskType& mask_type);
+    void parseCameraParameters(SE3 &extrinsics,
+                               In &intrinsics,
+                               DistortionModel &distortion_model,
+                               VectorX &distortion_coefficients,
+                               Vector2 &resolution,
+                               fp &timeshift_cam_imu,
+                               cv::Mat &mask,
+                               MaskType &mask_type);
 
-  /**
+    /**
    * @brief Parse the given origin (initial state) and the initial time
    *
    * @param T
    * @param b
    * @param t0
    */
-  void parseGivenOrigin(SE23& T, Vector6& b, fp& t0);
+    void parseGivenOrigin(SE23 &T, Vector6 &b, fp &t0);
 
-  /**
+    /**
    * @brief Parse the image preprocessing equalization method
    *
    * @param eq
    */
-  void parseEqualizationMethod(EqualizationMethod& eq);
+    void parseEqualizationMethod(EqualizationMethod &eq);
 
-  /**
+    /**
    * @brief Parse the feature representation
    *
    * @param rep
    */
-  void parseFeatureRepresentation(FeatureRepresentation& rep);
+    void parseFeatureRepresentation(FeatureRepresentation &rep);
 
-  /**
+    /**
    * @brief Parse the projection method
    *
    * @param proj
    */
-  void parseProjectionMethod(ProjectionMethod& proj);
+    void parseProjectionMethod(ProjectionMethod &proj);
 
-  /**
+    /**
    * @brief Parse the feature detector type
    *
    * @param detector
    */
-  void parseDetectorType(FeatureDetector& detector);
+    void parseDetectorType(FeatureDetector &detector);
 
-  /**
+    /**
    * @brief Parse the initial covariance
    *
    * @param D_cov Covariance of the D element of the MSCEqF state
@@ -240,9 +240,9 @@ class OptionParser
    * @param L_cov Covariance of the L element of the MSCEqF state (Identity by default)
    *
    */
-  void parseInitialCovariance(Matrix9& D_cov, Matrix6& delta_cov, Matrix6& E_cov, Matrix4& L_cov);
+    void parseInitialCovariance(Matrix9 &D_cov, Matrix6 &delta_cov, Matrix6 &E_cov, Matrix4 &L_cov);
 
-  /**
+    /**
    * @brief Parse the process noise
    *
    * @param w_std Standard deviation of the process noise of the angular velocity
@@ -250,28 +250,28 @@ class OptionParser
    * @param bw_std Standard deviation of the process noise of the gyroscope bias
    * @param ba_std Standard deviation of the process noise of the accelerometer bias
    */
-  void parseProcessNoise(fp& w_std, fp& a_std, fp& bw_std, fp& ba_std);
+    void parseProcessNoise(fp &w_std, fp &a_std, fp &bw_std, fp &ba_std);
 
-  /**
+    /**
    * @brief Parse the measurement noise, it transforms it in normalized pixel noise if intrinsics parameter are not
    * refined online.
    *
    * @param pix_std Standard deviation of the measurement noise of the pixel coordinates
    * @param opts State options
    */
-  void parsePixStd(fp& pix_std, const StateOptions& opts);
+    void parsePixStd(fp &pix_std, const StateOptions &opts);
 
-  /**
+    /**
    * @brief Parse the zero velocity update method
    *
    * @param zvu zero velocity update method
    */
-  void parseZeroVelocityUpdate(ZeroVelocityUpdate& zvu);
+    void parseZeroVelocityUpdate(ZeroVelocityUpdate &zvu);
 
-  YAML::Node node_;       //!< YAML node
-  std::string filepath_;  //!< filepath
-};
+    YAML::Node node_;      //!< YAML node
+    std::string filepath_; //!< filepath
+  };
 
-}  // namespace msceqf
+} // namespace msceqf
 
-#endif  // OPTIONS_HPP
+#endif // OPTIONS_HPP

